reflex/app.py

```diff
@@ -89,7 +97,7 @@ class App(Base):
     state: Type[State] = DefaultState
 
     # Class to manage many client states.
-    state_manager: StateManager = StateManager()
+    state_manager: StateManager = StateManagerMemory(state=DefaultState)
```

Comment:
Why do we set this if we're overriding it again in the `__init__`?

Reply:
I didn't want to make it `Optional` and default it to `None` because then we have to put `is not None` checks all over the place.

But if we do something like `StateManager.create(state=DefaultState)` here, _and_ redis is used, then we create an unserializable default, which pydantic doesn't like.

So the default `StateManager` is the `StateManagerMemory`, then inside `__init__`, we call `StateManager.create(...)` to update the reference to either memory or redis.